modules/apps/analytics/analytics-message-sender-api/src/main/java/com/liferay/analytics/message/sender/model/listener/BaseEntityModelListener.java

```diff
@@ -415,18 +416,11 @@ protected JSONObject serialize(
 				User user = (User)baseModel;
 
 				try {
-					List<Group> groups = user.getSiteGroups();
+					long[] groupIds = TransformUtil.transformToLongArray(
+						user.getSiteGroups(), Group::getGroupId);
 
-					if (!groups.isEmpty()) {
-						long[] membershipIds = new long[groups.size()];
-
-						for (int i = 0; i < groups.size(); i++) {
-							Group group = groups.get(i);
-
-							membershipIds[i] = group.getGroupId();
-						}
-
-						memberships.put(Group.class.getName(), membershipIds);
+					if (ArrayUtil.isNotEmpty(groupIds)) {
```

Comment:
@julschong From performance point of view I prefer the original if-logic. Check the list size first, so we don't go into the method calls (stacking pushing, try-block entering and exiting) and creating new arrays, when the list is empty.